### PR TITLE
fix: Keep track of ongoing uploads (WEBAPP-5901)

### PR DESCRIPTION
**Pull request by atomrc**

### CODE REVIEW

File: test/unit_tests/assets/AssetUploaderSpec.js [hunk 1]
```diff
@@ -0,0 +1,106 @@
+/*
+ * Wire
+ * Copyright (C) 2018 Wire Swiss GmbH
```
Comment (bennycode) [inline test/unit_tests/assets/AssetUploaderSpec.js]: 2019

File: src/script/util/dependenciesResolver.js [hunk 2]
```diff
@@ -46,8 +45,7 @@ const resolver = {
    */
   resolve: memoize(dependencyClass => {
     if (!dependencyGraph) {
-      logger.error('Failed to resolve dependency: no dependency graph set');
-      throw new Error('Cannot resolve dependency');
+      throw new Error(`Cannot resolve dependency ${dependencyClass}`);
```
Comment (bennycode) [inline src/script/util/dependenciesResolver.js]:
```suggestion
      throw new Error(`Cannot resolve dependency "${dependencyClass}"`);
```

File: src/script/components/asset/videoAsset.js [hunk 3]
```diff
@@ -17,10 +17,9 @@
  *
  */
 
-window.z = window.z || {};
-window.z.components = z.components || {};
+import AbstractAssetTransferStateTracker from './AbstractAssetTransferStateTracker';
 
-z.components.VideoAssetComponent = class VideoAssetComponent {
+class VideoAssetComponent extends AbstractAssetTransferStateTracker {
```
Comment (bennycode) [inline src/script/components/asset/videoAsset.js:22]: I love the name. 😍

File: src/script/components/asset/AbstractAssetTransferStateTracker.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
+/*
+ * Wire
+ * Copyright (C) 2018 Wire Swiss GmbH
```
Comment (bennycode) [inline src/script/components/asset/AbstractAssetTransferStateTracker.js]: 2019

File: src/script/assets/AssetUploader.js [hunk 5]
```diff
@@ -0,0 +1,97 @@
+/*
+ * Wire
+ * Copyright (C) 2018 Wire Swiss GmbH
```
Comment (bennycode) [inline src/script/assets/AssetUploader.js]: 2019

File: src/script/components/asset/videoAsset.js (hunk 3 above)
Comment (atomrc) [reply src/script/components/asset/videoAsset.js:22]: I could probably drop the `Abstract`, right? :D

**Pull request merged**